### PR TITLE
chore: move `start_server` util in tests into a fixture

### PR DESCRIPTION
**Pull request by dominikg**
so we don't have to try/finally inside the tests and leave playwright created contexrt alone

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Review by benmccann — CHANGES_REQUESTED** [review]
Some tests are failing with:
>  Test timeout of 45000ms exceeded while tearing down "start_server".

Also, I see `load does ignore ttl if fetch cache options says so` flaking again, which I haven't seen flake at all since `context.close()` was introduced. It's possible that we've just had a lucky run, but I'd want to see that it's failing in other tests before merging this

How do we know that `context.close()` will be run before `server.close()` with this PR?

**Comment by dominikg** [comment]
according to https://playwright.dev/docs/test-fixtures  , context is a fixture too, but one provided by playwright. the default page (another fixture) belongs to that context so i'd assume they are only closing the context after the page fixture

further down on that page they list the execution order https://playwright.dev/docs/test-fixtures#execution-order
additionally, fixtures can depend on each other (simply by using one fixture in another) so if teardown runs in reverse order and we need to ensure context is torn down before start_server, we'd have to make context depend on start_server? :thinking: arf

File: packages/kit/test/utils.js [hunk 1]
```diff
@@ -110,6 +110,90 @@ export const test = base.extend({
...
+	start_server: async ({}, use) => {
+		/**
+		 * @type {http.Server}
+		 */
+		let server;
+
+		/**
+		 * @type {Set<import('net').Socket>}
+		 */
+		let sockets;
```
Comment (benmccann) [inline packages/kit/test/utils.js:124]: it seemed to be working without all the socket craziness before. do we really need to add the extra complication?

Comment (dominikg) [reply packages/kit/test/utils.js:124]: this is what i found as advice for force closing http.Server.  I had logging locally and it actually does call socket.destroy in some cases, which should speed up the close at best and prevent timeouts at worst.

Given that this lives in a fixture and is out of the way for tests, not even used unless you actively call start_server(), i'd say it's worth it.

**Review by Rich-Harris — APPROVED** [review]

**Pull request merged**